sympy/solvers/solveset.py

```diff
@@ -1030,6 +1033,9 @@ def _solveset(f, symbol, domain, _check=False):
         if m not in {S.ComplexInfinity, S.Zero, S.Infinity,
                               S.NegativeInfinity}:
             f = a/m + h  # XXX condition `m != 0` should be added to soln
+        if isinstance(h, TrigonometricFunction) and domain.is_subset(S.Reals):
+            # solve this by inversion
+            linear_trig = True
```

Comment:
Second code change to allow problem to be solved by inversion, skipping the rewriting of `_solve_trig`.